api/spec/src/billing/invoices.tsp

```diff
@@ -0,0 +1,397 @@
+
+  /**
+   * Retry advancing the invoice's state to the next status.
+   *
+   * The call doesn't "approve the invoice", it only advances the invoice to the next status if the transition would be automatic.
+   *
+   * This call can be used to retry advancing the invoice's state to the next status if the previous attempt failed.
+   *
+   * This call is valid in the following invoice statuses:
+   * - `draft_sync_failed`
+   * - `issuing_failed`
+   * - `validation_failed`
+   */
+  @post
+  @route("/workflow/advance")
+  @summary("Advance the invoice's state to the next status")
+  @operationId("billingInvoiceWorkflowAdvance")
+  advanceWorkflow(
+    @path
+    customerIdOrKey: ULIDOrKey,
+
+    @path
+    invoiceId: ULID,
+  ): {
+    @statusCode _: 200;
+    @body body: Invoices.Invoice;
+  } | OpenMeter.NotFoundError | OpenMeter.CommonErrors;
+
+  // Line management
+
+  /**
+   * Delete an invoice line
+   */
+  @delete
+  @route("/lines/{lineId}")
+  @summary("Delete an invoice line")
+  @operationId("billingDeleteInvoiceLine")
+  deleteInvoiceLine(@path customerIdOrKey: ULIDOrKey, @path invoiceId: ULID, @path lineId: ULID): {
+    @statusCode _: 204;
+  } | OpenMeter.NotFoundError | OpenMeter.CommonErrors;
+
+  /**
+   * Update an invoice line
+   */
+  @put
+  @route("/lines/{lineId}")
+  @summary("Update an invoice line")
+  @operationId("billingUpdateInvoiceLine")
+  updateInvoiceLine(
+    @path customerIdOrKey: ULIDOrKey,
+    @path invoiceId: ULID,
+    @path lineId: ULID,
+    @body request: Invoices.InvoiceLine,
+  ): {
+    @statusCode _: 200;
+    @body body: Invoices.InvoiceLine;
+  } | OpenMeter.NotFoundError | OpenMeter.CommonErrors;
+
+  /**
+   * TODOs:
+   * - min/max spend support
+   * - discounts
+   * - tax support (detailed)
+   */
+}
+
+/**
+ * InvoiceOrderBy specifies the ordering options for invoice listing.
+ */
+@friendlyName("BillingInvoiceOrderBy")
+enum InvoiceOrderBy {
+  customerName: "customer_name",
+  issuedAt: "issued_at",
+  status: "status",
+  createdAt: "created_at",
+  updatedAt: "updated_at",
+}
+
+/**
+ * Response for creating a pending charge
+ */
+@friendlyName("BillingCreatePendingItemResponse")
+model CreatePendingItemResponse {
+  @summary("The created pending charge")
+  item: Invoices.InvoiceLine;
+
+  @summary("The invoice the item was added to")
+  invoice: Invoices.Invoice;
+}
+
+/**
+ * Request to void an invoice
+ */
+@friendlyName("BillingVoidInvoiceInput")
+model VoidInvoiceInput {
+  /**
+   * The action to take on the voided items.
+   */
+  @summary("The action to take on the voided items")
+  action: VoidInvoiceItemAction;
+
+  /**
+   * The reason for voiding the invoice.
+   */
+  @summary("The reason for voiding the invoice")
+  reason: string;
+
+  /**
+   * Per line item overrides for the action.
+   *
+   * If not specified, the `action` will be applied to all items.
+   */
+  itemOverrides: VoidInvoiceItemOverride[];
```

Comment:
Should this be a nullable field? I know in Go empty array and null is similar but I think in JSON `null` is more expressive